osu.Game.Rulesets.Osu.Tests/TestSceneSpinnerInput.cs

```diff
@@ -194,11 +196,11 @@ public void TestRewind()
             assertTotalRotation(3750, 810);
             assertTotalRotation(3500, 720);
             assertTotalRotation(3250, 530);
-            assertTotalRotation(3000, 540);
+            assertTotalRotation(3000, 450);
```

Comment:
Based on our discussions and the 3 implementations I've gone through, this is not correct. Can you explain this for me?

Up to this point, in normal playback, we've got:
1 CW spin
+ 0.5 CCW spins
+ 0.25 CW spins

Which sums up to be 1CW spin + 0.5CCW spin - the 0.25 is irrelevant because it's "returning to 0". In other words, this should be 540 as far as I can tell.

Reply:
I'll go through the test discrepancies, but on a quick check you're likely right. I mentioned in the OP that i didn't really go over tests with a fine comb, just made the ones you wrote pass.

That said, I don't think this is an issue with the algorithm. Ignoring the test failures, It'd be helpful to know if you're on board with the algorithm itself so I can either prioritise fixing the tests to get it in for this release, or hold it back to give us more time to discuss.

Reply:
Before I go into the algorithm, have you deviated from the original goal of requiring crossing 0 that resulted in these tests being the way they are now?

Reply:
No, I don't believe so, that part should still be the same.

Most of the change is in the handling of rewind (simplifying it).